RecoLocalTracker/SiPixelClusterizer/plugins/PixelThresholdClusterizer.cc

```diff
@@ -471,8 +476,8 @@ SiPixelCluster PixelThresholdClusterizer::make_cluster(const SiPixelCluster::Pix
 	      
 	      SiPixelCluster::PixelPos newpix(r,c);
 	      if(!doSplitClusters){
-	      
-	      cluster.add(newpix, theBuffer(r,c));}
+
+	      cluster.add(newpix, std::min(theBuffer(r, c), int(std::numeric_limits<uint16_t>::max())));}
```

Comment:
If the values in `theBuffer` _cannot_ be larger than `65535`, wouldn't it be better to change `SiPixelArrayBuffer::operator()(int row, int col)` to return a `uint16_t`, rather than adding the check here and potentially everywhere else it is used ?

If the valkues in `theBuffer` _can_ be larger than `65535`, why do they need to be clamped ?

Reply:
(I'll simply answer on the technical part; obviously, I'm not a Pixel expert).

In the current implementation, and before this PR, the values to and from `theBuffer` can be larger than 65535. `theBuffer` (instance of `SiPixelArrayBuffer`) only deals with `int`s.

The clamping avoids silent value changes in type conversion when, inside `PixelThresholdClusterizer`, values are passed from `theBuffer` to instances of [`AccretionCluster`](https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/RecoLocalTracker/SiPixelClusterizer/plugins/PixelClusterizerBase.h#L47), as the latter uses `uint16_t`. Relevant to this discussion is also the fact that `uint16_t` is what is used by in the data format [`SiPixelCluster`](https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/SiPixelCluster/interface/SiPixelCluster.h#L202) (notice the implicit conversion in [this constructor](https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/DataFormats/SiPixelCluster/interface/SiPixelCluster.h#L33)).

> rather than adding the check here and potentially everywhere else it is used ?

That's why I originally thought it was safer to modify `SiPixelArrayBuffer` rather than its "clients". In fact, I realise now that this PR would have to be extended to [`PixelThresholdClusterizerForBricked`](https://github.com/cms-sw/cmssw/blob/master/RecoLocalTracker/SiPixelClusterizer/plugins/PixelThresholdClusterizerForBricked.cc#L135) (a 2nd client of `SiPixelArrayBuffer`) (although, maybe bricked pixels are getting dropped, https://github.com/cms-sw/cmssw/pull/38061).